atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -278,6 +282,34 @@ protected CassandraKeyValueServiceImpl(Logger log,
         this.getCandidateCellsForSweepingImpl = new CassandraGetCandidateCellsForSweepingImpl(cellPager);
     }
 
+    /**
+     * Creates a thread pool that maintains between (poolSize * numServers) and (maxConnectionBurstSize * numServers)
+     * threads. When the max pool size is hit, new requests will not be queued but rather will be run in the caller's
+     * thread. Throttling will be taken care of by the underlying client pools.
+     */
+    private static ExecutorService createExecutor(CassandraKeyValueServiceConfig config) {
+        int numServers = config.servers().size();
+        return PTExecutors.newThreadPoolExecutor(
```

Comment:
Should we instrument this now that its growing? I'm working on some separate metrics to better instrument executors & backing queue, so I can do in a follow up PR if desired.

```
        return new InstrumentedExecutorService(
                PTExecutors.newThreadPoolExecutor(...),
                AtlasDbMetrics.getMetricRegistry(),
                "atlasdb-cassandra-kvs");
```